apps/angular-store-app/src/app/cart/state/cart.effects.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { Actions, Effect, ofType } from '@datorama/akita-ng-effects';
+import { CartActions } from './cart.actions';
+import { map, switchMap, tap } from 'rxjs/operators';
+import { CartStore } from './cart.store';
+import { of, timer } from 'rxjs';
+
+@Injectable()
+export class CartEffects {
+  constructor(private actions$: Actions, private cartStore: CartStore) {
+    console.log(this);
+  }
+
+  @Effect()
+  removeItem$ = this.actions$.pipe(
+    ofType(CartActions.removeItem),
+    tap((val) => console.log('[Cart Effects] removeItem$', val)),
+    switchMap((test) => {
+      // do async operations
+      this.cartStore.remove(test.productId);
+      return of(this.actions$.dispatch(CartActions.removeItemSuccess()));
+    })
+  );
+
+  @Effect()
```

Comment:
Use the `createEffect` function in one of these methods, please.

Reply:
Changed.